lib/resources/shadow.rb

```diff
@@ -0,0 +1,132 @@
+# encoding: utf-8
+# copyright: 2016, Chef Software Inc.
+# author: Dominik Richter
+# author: Christoph Hartmann
+
+# The file format consists of
+# - user
+# - password
+# - last_change
+# - min_days before password change
+# - max_days until password change
+# - warn_days before warning about expiry
+# - inactive_days before deactivating the account
+# - expiry_date when this account will expire
+
+class Shadow < Inspec.resource(1)
+  name 'shadow'
+  desc 'Use the shadow InSpec resource to test the contents of /etc/shadow, '\
+       'which contains the following information for users that may log into '\
+       'the system and/or as users that own running processes.'
+  example "
+    describe shadow do
+      its('users') { should_not include 'forbidden_user' }
+    end
+
+    describe shadow.users('bin') do
+      its('password') { should cmp 'x' }
+      its('count') { should eq 1 }
+    end
+  "
+
+  attr_reader :params
+  attr_reader :content
+  attr_reader :lines
+
+  def initialize(path = '/etc/shadow', opts = nil)
+    opts ||= {}
+    @path = path || '/etc/shadow'
```

Comment:
Almost: what if `shadow(nil)`. It will not happen intentionally, but may happen by accident. I'm not sure yet how we want to handle this case though. Throwing a ruby fail seems to harsh, maybe improve this in the next iteration. It is true for a lot of resources...